types/three/src/core/BufferGeometry.d.ts

```diff
@@ -91,7 +92,10 @@ export class BufferGeometry extends EventDispatcher {
     getIndex(): BufferAttribute | null;
     setIndex(index: BufferAttribute | number[] | null): BufferGeometry;
 
-    setAttribute(name: string, attribute: BufferAttribute | InterleavedBufferAttribute): BufferGeometry;
+    setAttribute(
+        name: BuiltinShaderAttributeName | (string & {}),
```

Comment:
why is this `string & {}` I didn't think a name could be an object?

Reply:
that's hack to prevent conversion from `"const-string" | string` to `string`.

related: https://github.com/microsoft/TypeScript/issues/29729#issuecomment-505826972

Reply:
hm, im not comfortable with this hack, it could bring more problems than solutions – https://github.com/microsoft/TypeScript/issues/29729#issuecomment-706401313

Reply:
I think those problems are not related with this usecase, but I'm not familiar with those corner cases, so I might have missed something. 

description:
- Type guards don't narrow the variable
  - In `.d.ts` we dont need type guards because `.d.ts` doesn't including implementation.
- No property auto-completion when an object type is in the union
  - it shouldn't occur accept objects as attribute name
- No string auto-completion when extended by generic type
  - thats maybe problem for future, but currently, thats not problem because we don't using generics for attribute name
- Interferes with conditional types that expect `string & {}` to not be a subtype of `object`
  - anyone want to check `name extends object`?

Reply:
Okay well please write a test for this feature as requested we can then be sure that the issues above don't relate to what we're doing now.